sanityv3/schemas/objects/teaser.tsx

```diff
@@ -87,14 +111,35 @@ export default {
         input: CompactBlockEditor,
       },
       of: [titleContentType],
+      hidden: ({ parent }: TeaserDocument) => parent.isBigText,
     },
     {
       name: 'text',
       title: 'Text content',
       type: 'array',
       of: [blockContentType],
       validation: (Rule: Rule) =>
-        Rule.custom((value: PortableTextBlock[]) => validateCharCounterEditor(value, 600)).warning(),
+        Rule.custom((value: PortableTextBlock[], ctx: ValidationContext) => {
+          if (!(ctx.parent as Teaser)?.isBigText) {
+            return validateCharCounterEditor(value, 600)
+          }
+          return true
+        }).warning(),
+      hidden: ({ parent }: TeaserDocument) => parent.isBigText,
+    },
+    {
+      name: 'bigText',
+      title: 'Text content',
+      type: 'array',
+      of: [blockContentTypeForBigText],
+      validation: (Rule: Rule) =>
+        Rule.custom((value: PortableTextBlock[], ctx: ValidationContext) => {
+          if ((ctx.parent as Teaser)?.isBigText) {
+            return validateCharCounterEditor(value, 600)
+          }
+          return true
+        }),
+      hidden: ({ parent }: TeaserDocument) => !parent.isBigText,
     },
```

Comment:
Can't we use the `Rule.custom((value: PortableTextBlock[]) => validateCharCounterEditor(value, 600)).warning()` and 
`Rule.custom((value: PortableTextBlock[]) => validateCharCounterEditor(value, 600))` for text and bigText.. seems like no difference to me.

Reply:
one is a warning, the other is required